### PR TITLE
June 3 Stability and Performance Enhancements

### PR DESCRIPTION
**Pull request by davmillar**
- Improves build workflow for app files.
- Removes old or outdated content.
- Improves styling and layout of ancillary pages.
- Removes some of the CSS gradients and partially-transparent borders to improve usability and perforamnce.
- Admin area improvements.
- Additional code cleanup.

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -4,25 +4,25 @@ RewriteEngine on
 
 RewriteBase /
 
-RewriteCond %{HTTP_HOST} ^www\.davesmapper\.com [NC]
-RewriteRule ^(.*)$ http://davesmapper.com/$1 [L,R=301]
+RewriteCond %{HTTP_HOST} ^www\.mapper\.localhost [NC]
+RewriteRule ^(.*)$ http://mapper.localhost/$1 [R=301,QSA]
```
Comment (morgajel) [inline .htaccess]: localhost? is that *really* what you intended?

File: admin/index.php [hunk 2]
```diff
@@ -90,7 +90,14 @@ function bounceOut(){ header("Location: http://davesmapper.com/admin"); exit; }
...
 		mysql_query("INSERT INTO artists (name, url_slug, initials, icon, url, bio, email, password)
-								 VALUES ('".mysql_real_escape_string($_POST['aname'])."', '".mysql_real_escape_string($_POST['aurlslug'])."', '".mysql_real_escape_string($_POST['ainit'])."', '".mysql_real_escape_string($_POST['aicon'])."', '".mysql_real_escape_string($_POST['alink'])."', '".mysql_real_escape_string($_POST['abio'])."', '".mysql_real_escape_string($_POST['aemail'])."', '".sha1($_POST['aemail']."bacon"."c00lt1l3sbr0")."')");
+								 VALUES ('".mysql_real_escape_string($_POST['aname'])."',
+								 				 '".mysql_real_escape_string($_POST['aurlslug'])."',
+								 				 '".mysql_real_escape_string($_POST['ainit'])."',
+								 				 '".mysql_real_escape_string($_POST['aicon'])."',
+								 				 '".mysql_real_escape_string($_POST['alink'])."',
+								 				 '".mysql_real_escape_string($_POST['abio'])."',
+								 				 '".mysql_real_escape_string($_POST['aemail'])."',
+								 				 '".sha1($_POST['aemail']."bacon".$_POST['apass'])."')");
```
Comment (morgajel) [inline admin/index.php]: I think you can reduce the horizontal spacing on these a little bit to keep them from scrolling off the screen when viewing the diff on github. Don't worry, there's a lot of bad ways to do it, a few mediocre ways, and no good ways :)

File: admin/index.php [hunk 3]
```diff
@@ -326,22 +333,27 @@ function bounceOut(){ header("Location: http://davesmapper.com/admin"); exit; }
...
 										<input type="text" name="alink" id="alink" autocomplete="off" />
-										<input type="text" name="aemail" id="aemail" autocomplete="off" />
 										<input type="text" name="aicon" id="aicon" class="little" autocomplete="off" /><br/>
 											<label for="alink" class="labelTxt tBox">Link URL</label>
-											<label for="aemail" class="labelTxt tBox">Email</label>
 											<label for="aicon" class="labelTxt nBox">.png</label>
 									</div>
 									<div class="fieldset">
+										<h4 class="legend">Auth</h4>
+										<input type="text" name="aemail" id="aemail" autocomplete="off" />
```
Comment (morgajel) [inline admin/index.php]: why is autocomplete off on email? that seems... obnoxious.

File: build.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+mkdir -p ./assets/css/
+rm -r ./assets/js/
+mkdir -p ./assets/js/
+
+echo "Combining and compressing main CSS...";
+
+cat \
+  style/style.less \
+  | lessc --clean-css - \
+  | yui-compressor --type=css \
```
Comment (morgajel) [inline build.sh]: :+1:

File: supporters.php [hunk 5]
```diff
@@ -100,34 +97,7 @@
...
-							<h3>James G.</h3>
-							<p>Mushroom Swissburger from Checkers.</p></section>
-						<section class="keyimg"><p>Uneaten burger donations from Daniel S. (x2), Ryan S., Aron S., and Aaron P.</p></section>
-						<section class="keyimg"><img src="images/sandwiches/1.png" alt="Sub Sandwich 1" />
-							<h3>Alexander B.</h3>
-							<p>Veggie with Guacamole on Honey Wheat from Our Town Deli</p></section>
-						<section class="keyimg"><p>Uneaten sandwich donations from R. Lang, T K Backman , Alan D., and Forrest H.</p></section>
-					</section>
-					<p>As of 2012, I'm no longer asking for sandwich donations. Instead, I'm asking for any amount you'd like to give toward my personal medical expenses and/or donations toward the Crohn's and Colitis Foundation of America. Links for those are in the footer.</p>
+					<p>If you would like to help support the mapper, please consider donations toward Dave's personal medical bills or a donation to the Crohn's and Colitis Foundation of America. Links for those are in the footer.</p>
```
Comment (morgajel) [inline supporters.php]: Not to make light of your condition, but there is a certain irony in this change.

**Comment by matthewpucc** [comment]
You really should use prepared statements with mysql instead of inlining some real escape strings.

**Comment by matthewpucc** [comment]
also sha1 is lame. use bcrypt.

File: .htaccess (hunk 1 above)
Comment (davmillar) [inline .htaccess]: Whoops - my bad. How did that sneak in?

File: admin/index.php (hunk 2 above)
Comment (davmillar) [inline admin/index.php]: Valid concern. :)

File: admin/index.php (hunk 3 above)
Comment (davmillar) [inline admin/index.php]: This is a form for adding a new user, so I wouldn't be autocompleting with my own email address. Once I build in the part for users to create their own accounts, that form will not have autocomplete turned off.

**Comment by davmillar** [comment]
Will add those to my list of things to learn - thanks @matthewpucc

File: supporters.php (hunk 5 above)
Comment (davmillar) [inline supporters.php]: Indeed. But the main reason for this changes is that it's not all about me. This section will get updated again soon to add Patreon info for many of the mapper contributors and ways to support the other folks who've put so much into making things for the mapper.

File: admin/index.php (hunk 3 above)
Comment (morgajel) [inline admin/index.php]: aaah, ok, I thought this was for signup stuff, heh. I didn't think it through.

File: admin/index.php (hunk 2 above)
Comment (davmillar) [inline admin/index.php]: Just realized the over-indentation was because indentation was done with tabs instead of spaces. Fixed that in my editor :)

**Pull request merged**